### ISSUE TITLE
Test routine only logs metric of last dataloader item ?

### ISSUE THREAD

**Issue by darius522** [NoFish-528/encodec-pytorch#18]
https://github.com/ZhikangNiu/encodec-pytorch/blob/036c443e9f49bfa8967c13fc9c078b39ac2a9471/train_multi_gpu.py#L127-L128

Hey there! Thanks again for your effort on this repo. Very minor detail; it looks like the test function currently only logs the metrics for the last dataloader item (maybe I misunderstood how this plays out with DataParallel, if so my apology). Any reason behind that ?

**Comment by ZhikangNiu** [comment]
yes, it only log the metrics for the last dataloader item (in rank 0 when use ddp). You can feel free to change it :)

**Comment by darius522** [comment]
Thanks for clarifying, just wanted to ensure I understood that right.
